jaxley/modules/base.py

```diff
@@ -912,21 +913,65 @@ def step(
                 branchpoint_group_inds=self.branchpoint_group_inds,
                 debug_states=self.debug_states,
             )
-        else:
+            u["v"] = new_voltages.ravel(order="C")
+        elif solver == "fwd_euler":
             new_voltages = step_voltage_explicit(
-                voltages,
-                (v_terms + syn_v_terms) / cm,
-                (const_terms + i_ext + syn_const_terms) / cm,
-                coupling_conds_bwd=params["coupling_conds_bwd"],
-                coupling_conds_fwd=params["coupling_conds_fwd"],
-                branch_cond_fwd=params["branch_conds_fwd"],
-                branch_cond_bwd=params["branch_conds_bwd"],
+                voltages=voltages,
+                voltage_terms=(v_terms + syn_v_terms) / cm,
+                constant_terms=(const_terms + i_ext + syn_const_terms) / cm,
+                coupling_conds_upper=params["branch_uppers"],
+                coupling_conds_lower=params["branch_lowers"],
+                summed_coupling_conds=params["branch_diags"],
+                branchpoint_conds_children=params["branchpoint_conds_children"],
+                branchpoint_conds_parents=params["branchpoint_conds_parents"],
+                branchpoint_weights_children=params["branchpoint_weights_children"],
+                branchpoint_weights_parents=params["branchpoint_weights_parents"],
+                par_inds=self.par_inds,
+                child_inds=self.child_inds,
                 nbranches=self.total_nbranches,
-                parents=self.comb_parents,
+                solver=voltage_solver,
                 delta_t=delta_t,
+                children_in_level=self.children_in_level,
+                parents_in_level=self.parents_in_level,
+                root_inds=self.root_inds,
+                branchpoint_group_inds=self.branchpoint_group_inds,
+                debug_states=self.debug_states,
```

Comment:
I think you could condense this down to sth like, since the step_voltage functions take the same kwargs, they dont have to be repeated.

```python

solver_kwargs = {
    "voltages": voltages,
    "voltage_terms": (v_terms + syn_v_terms) / cm,
    "constant_terms": (const_terms + i_ext + syn_const_terms) / cm,
    "coupling_conds_upper": params["branch_uppers"],
    "coupling_conds_lower": params["branch_lowers"],
    "summed_coupling_conds": params["branch_diags"],
    "branchpoint_conds_children": params["branchpoint_conds_children"],
    "branchpoint_conds_parents": params["branchpoint_conds_parents"],
    "branchpoint_weights_children": params["branchpoint_weights_children"],
    "branchpoint_weights_parents": params["branchpoint_weights_parents"],
    "par_inds": self.par_inds,
    "child_inds": self.child_inds,
    "nbranches": self.total_nbranches,
    **remaining_kwargs
    }

if solver == "bwd_euler":
    new_voltages = step_voltage_implicit(voltages, dt, **solver_kwargs)
    u["v"] = new_voltages.ravel(order="C")
elif solver == "fwd_euler":
    new_voltages = step_voltage_explicit(voltages, dt, **solver_kwargs)
    u["v"] = new_voltages.ravel(order="C")
elif solver == "crank_nicolson":
    new_voltages = step_voltage_implicit(voltages, dt, **solver_kwargs)
    u["v"] = 2 * half_step_voltages.ravel(order="C") - voltages
else:
    raise ValueError(f"Unknown solver {solver}")
```